### PR TITLE
Inject app. arguments into dispatched functions + exit codes + pep8

### PR DESCRIPTION
**Pull request by nielsvm**
**THIS CODE IS UNTESTED - TEST FIRST**

Having that said, I expect perhaps only a subtle thing, it should all work:
```
pylint app/arguments.py app/containers.py app/orchestration.py
************* Module orchestration
app/orchestration.py:103:6: W0511: TODO: actually enforce arbitrary docroots through positional arguments (fixme)

------------------------------------------------------------------
Your code has been rated at 9.95/10 (previous run: 9.95/10, +0.00)
```

See commit messages:

```
commit b1f0131a753d8d87aaa92f440a823c7257bfb907
    Since functions are objects, I'm passing them as constants into
ARGS.mainfunc. Once the command gets called, it always passes
ARGS so that it becomes available everywhere in the codebase. Each
ran command must now also state their exit status.
```
```
commit 2ab1e79c763d90268f075087fa82fe7d0edf30ca
    ARGS becomes the injected 'args' on each function, which
now all return args.exit_ok assuming they didn't fail. If commands
can still fail, over time you should return exit_fail instead.
```
```
commit e16c9e610b3fda3d623f5cfaeb4b531d623230af (HEAD -> appargs, origin/appargs)
    Same for app/orchestration.py, which I also made PEP8 compliant.
```

### CODE REVIEW

**Comment by nielsvm** [comment]
I expect Travis still to fail for that `@TODO` in the files I touched - but not for more things.

File: app/arguments.py [hunk 1]
```diff
@@ -1,113 +1,95 @@
 #!/usr/bin/env python3
-"""Allows to pass arguments to the drucker command"""
+"""Parsing of command line arguments passed to the drucker."""
```
Comment (anavarre) [inline app/arguments.py]: It's `drucker` only.

File: app/arguments.py [hunk 2]
```diff
@@ -1,113 +1,95 @@
...
 
-import subprocess as s
+import sys
 import argparse
-import colorful as c
-import variables as v
-import orchestration as o
+import subprocess
+import colorful
+import variables
```
Comment (anavarre) [inline app/arguments.py]: `variables as vars` looks better

File: app/arguments.py [hunk 3]
```diff
@@ -1,113 +1,95 @@
...
+"""Parsing of command line arguments passed to the drucker."""
 
-import subprocess as s
+import sys
 import argparse
-import colorful as c
-import variables as v
-import orchestration as o
+import subprocess
+import colorful
```
Comment (anavarre) [inline app/arguments.py]: `colorful as color`

File: app/arguments.py [hunk 4]
```diff
@@ -1,113 +1,95 @@
...
+import variables
 import containers
+import orchestration
```
Comment (anavarre) [inline app/arguments.py]: `orchestration as orch`

File: app/arguments.py [hunk 5]
```diff
@@ -1,113 +1,95 @@
...
-"""Allows to pass arguments to the drucker command"""
+"""Parsing of command line arguments passed to the drucker."""
 
-import subprocess as s
+import sys
 import argparse
-import colorful as c
-import variables as v
-import orchestration as o
+import subprocess
```
Comment (anavarre) [inline app/arguments.py]: `subprocess as sub`

File: app/arguments.py [hunk 6]
```diff
@@ -1,113 +1,95 @@
...
 
 
-def return_version():
+def return_version(args):
     """Returns either the latest commit hash or tagged release"""
-    latest_commit = s.getoutput("cd %s && git rev-parse --short HEAD" % (v.APP_ROOT))
-
-    if 'dev' in v.APP_VERSION:
-        print("You are running the dev version at commit " + c.orange(latest_commit))
+    latest_commit = subprocess.getoutput("cd %s && git rev-parse --short HEAD"
```
Comment (anavarre) [inline app/arguments.py]: `sub.getoutput`

File: app/arguments.py [hunk 7]
```diff
@@ -1,113 +1,95 @@
...
+                                         % (variables.APP_ROOT))
+    if 'dev' in variables.APP_VERSION:
```
Comment (anavarre) [inline app/arguments.py]: `vars.APP_VERSION`

File: app/arguments.py [hunk 8]
```diff
@@ -1,113 +1,95 @@
...
 
-def return_version():
+def return_version(args):
     """Returns either the latest commit hash or tagged release"""
-    latest_commit = s.getoutput("cd %s && git rev-parse --short HEAD" % (v.APP_ROOT))
-
-    if 'dev' in v.APP_VERSION:
-        print("You are running the dev version at commit " + c.orange(latest_commit))
+    latest_commit = subprocess.getoutput("cd %s && git rev-parse --short HEAD"
+                                         % (variables.APP_ROOT))
```
Comment (anavarre) [inline app/arguments.py]: `vars.APP_ROOT`

File: app/arguments.py [hunk 9]
```diff
@@ -1,113 +1,95 @@
...
+    if 'dev' in variables.APP_VERSION:
+        ver = colorful.orange(latest_commit)
```
Comment (anavarre) [inline app/arguments.py]: `hash = color.orange(latest_commit)`

File: app/arguments.py [hunk 10]
```diff
@@ -1,113 +1,95 @@
...
+        print("You are running the dev version at commit " + ver)
```
Comment (anavarre) [inline app/arguments.py]: s/`ver`/`hash`

File: app/arguments.py [hunk 11]
```diff
@@ -1,113 +1,95 @@
...
     else:
-        print("You are running tagged release " + c.orange(v.APP_VERSION))
+        ver = colorful.orange(variables.APP_VERSION)
```
Comment (anavarre) [inline app/arguments.py]: `tag = color.orange(vars.APP_VERSION)`

File: app/arguments.py [hunk 12]
```diff
@@ -1,113 +1,95 @@
...
+        print("You are running tagged release " + ver)
```
Comment (anavarre) [inline app/arguments.py]: s/`ver`/`tag`

**Review by anavarre — COMMENTED** [review]
Initial and early review

File: app/arguments.py (hunk 5 above)
Comment (nielsvm) [reply app/arguments.py]: I recommend to not do this, while its totally common to alias your own modules, it becomes less conventient for third-party developers to not be able to recognize libraries like `subprocess` and `colorful`. Its better to only do this for your own modules, but obviously your choice :-)

**Comment by anavarre** [comment]
Commands work and we've worked towards fixing most bugs. Let's get this in and continue onto refactoring `drucker.sh`

**Pull request merged**